modules/sharedaddy/recaptcha.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+/**
+ * Class that handles reCAPTCHA.
+ */
+class Jetpack_ReCaptcha {
+
+	/**
+	 * URL to which requests are POSTed.
+	 *
+	 * @const string
+	 */
+	const VERIFY_URL = 'https://www.google.com/recaptcha/api/siteverify';
+
+	/**
+	 * Site key to use in HTML code.
+	 *
+	 * @var string
+	 */
+	private $site_key;
+
+	/**
+	 * Shared secret for the site.
+	 *
+	 * @var string
+	 */
+	private $secret_key;
+
+	/**
+	 * Config for reCAPTCHA instance.
+	 *
+	 * @var array
+	 */
+	private $config;
+
+	/**
+	 * Error codes returned from reCAPTCHA API.
+	 *
+	 * @see https://developers.google.com/recaptcha/docs/verify
+	 *
+	 * @var array
+	 */
+	private $error_codes = array(
+		'missing-input-secret'   => 'The secret parameter is missing',
+		'invalid-input-secret'   => 'The secret parameter is invalid or malformed',
+		'missing-input-response' => 'The response parameter is missing',
+		'invalid-input-response' => 'The response parameter is invalid or malformed',
+	);
+
+	/**
+	 * Create a configured instance to use the reCAPTCHA service.
+	 *
+	 * @param string $site_key   Site key to use in HTML code.
+	 * @param string $secret_key Shared secret between site and reCAPTCHA server.
+	 * @param array  $config     Config array to optionally configure reCAPTCHA instance.
+	 */
+	public function __construct( $site_key, $secret_key, $config = array() ) {
+		$this->site_key   = $site_key;
+		$this->secret_key = $secret_key;
+		$this->config     = wp_parse_args( $config, $this->get_default_config() );
+	}
+
+	/**
+	 * Get default config for this reCAPTCHA instance.
+	 *
+	 * @return array Default config
+	 */
+	public function get_default_config() {
```

Comment:
Same reason I made `get_verify_request_params` visibility to `public` &mdash; to have unit test. If there's good reason to mark it as private let me know and I'm happy to remove relevant unit test.